src/actions/authActions.js

```diff
@@ -346,8 +347,7 @@ export const checkAuthAction = (
         const saltedPin = await getSaltedPin(pin, dispatch);
         wallet = await decryptWallet(encryptedWallet, saltedPin, options);
       } else if (privateKey) {
-        const walletAddress = normalizeWalletAddress(encryptedWallet.address);
-        wallet = { ...encryptedWallet, privateKey, address: walletAddress };
+        wallet = new ethers.Wallet(privateKey);
```

Comment:
1) why we didn't change this before? :D 
2) `wallet = await decryptWallet(encryptedWallet, saltedPin, options);` additionally connects to the provider, while your approach will simply return the wallet instance. Do you think it's enough?
As I remember we're checking for the provider in other places, so it could be ok

Reply:
1. No idea 🤷‍♀️ I just found out that it was introduced a few months ago in this [PR](https://github.com/pillarwallet/pillarwallet/pull/1366) by @poocart. Maybe back then it was making sense and over the time the usage of these methods changed.
2. Yes, we're always connecting to provider first.

Reply:
1) Right, before it was enough to have this version `wallet = { ...encryptedWallet, privateKey, address: walletAddress };` since we just needed to actually know the privateKey and the wallet's address in the code below. Things have changed since then :)
2) we should fix the way we connect to the provider, it could be done in a separate PR. I've checked all the occurrences of that provider's connection and only 1 out of 7 is correct.
The problem is that `wallet.connect(provider)` will return a new instance which we don't store anywhere